class2.1/task3a.txt

```diff
@@ -0,0 +1,36 @@
+Test Case 1: New User
+
+Input: Program run with name "Alice".
+Expected Output: "Welcome, Alice!"
+
+Test Case 2: Returning User
+
+Input: Program run with name "Alice".
+Expected Output: "Hello again(x1), Alice"
+
+Test Case 3: Multiple Runs with Different Names
+
+Input: Program run with name "Bob".
```

Comment:
test cases looks good while you should also specify preconditions of test to ensure that the state of system is the same for any test run. In this case we should ensure that our history file has certain info - for example file should be deleted before test run to emulate situation of very first start. Or we can create file with certain data to emulate the situation when there were some user actions before.
Also you can add the exact command to your steps like and add step number to make it more readable for example
Step 1. Run program with name "Bob" using command `./hello Bob` .
Why it is important? it allows to save a time on test run as qa engineer will be able to copy the command instead of manual typing. It is very helpful when you run a lot of test cases every day.